homeassistant/components/flo/device.py

```diff
@@ -0,0 +1,176 @@
+        self._temperature: float = None
+        self._consumption_today: float = None
+
+        super().__init__(
+            hass,
+            _LOGGER,
+            name=f"{FLO_DOMAIN}-{device_id}",
+            update_interval=timedelta(seconds=60),
+        )
+
+    async def _async_update_data(self):
+        """Update data via library."""
+        try:
+            async with timeout(10):
+                await asyncio.gather(
+                    *[self._update_device(), self._update_consumption_data()]
+                )
+        except (RequestError) as error:
+            raise UpdateFailed(error)
+
+    @property
+    def location_id(self) -> str:
+        """Return Flo location id."""
+        return self._flo_location_id
+
+    @property
+    def id(self) -> str:
+        """Return Flo device id."""
+        return self._flo_device_id
+
+    @property
+    def device_name(self) -> str:
+        """Return device name."""
+        return f"{self.manufacturer} {self.model}"
+
+    @property
+    def mac_address(self) -> str:
+        """Return ieee address for device."""
+        return self._mac_address
+
+    @property
+    def manufacturer(self) -> str:
+        """Return manufacturer for device."""
+        return self._manufacturer
+
+    @property
+    def model(self) -> str:
+        """Return model for device."""
+        return self._model
+
+    @property
+    def rssi(self) -> float:
+        """Return rssi for device."""
+        return self._rssi
+
+    @property
+    def last_heard_from_time(self) -> str:
+        """Return lastHeardFromTime for device."""
+        return self._last_heard_from_time
+
+    @property
+    def device_type(self) -> str:
+        """Return the device type for the device."""
+        return self._device_type
+
+    @property
+    def available(self) -> bool:
+        """Return True if device is available."""
+        return self._available
+
+    @property
+    def current_system_mode(self) -> str:
+        """Return the current system mode."""
+        return self._current_system_mode
+
+    @property
+    def target_system_mode(self) -> str:
+        """Return the target system mode."""
+        return self._target_system_mode
+
+    @property
+    def current_flow_rate(self) -> float:
+        """Return current flow rate in gpm."""
+        return self._current_flow_rate
+
+    @property
+    def current_psi(self) -> float:
+        """Return the current pressure in psi."""
+        return self._current_psi
+
+    @property
+    def temperature(self) -> float:
+        """Return the current temperature in degrees F."""
+        return self._temperature
+
+    @property
+    def consumption_today(self) -> float:
+        """Return the current consumption for today in gallons."""
+        return self._consumption_today
+
+    @property
+    def firmware_version(self) -> str:
+        """Return the firmware version for the device."""
+        return self._firmware_version
```

Comment:
```suggestion
        return self._device_information["fwVersion"]
```

and then in `_update_device` you just store the device information on the instance.

Reply:
updated